include/zephyr/devicetree/port-endpoint.h

```diff
@@ -0,0 +1,267 @@
+/**
+ * @file
+ * @brief Port / Endpoint Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright 2024 NXP
+ * Copyright (c) 2024 tinyVision.ai Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_PORT_ENDPOINT_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_PORT_ENDPOINT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup devicetree-port-endpoint Devicetree Port Endpoint API
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Helper for @ref DT_INST_PORT_BY_ID
+ *
+ * This behaves the same way as @ref DT_INST_PORT_BY_ID but does not work if there is only
+ * a single port without address.
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @return port node associated with @p pid
+ */
+#define _DT_INST_PORT_BY_ID(inst, pid)                                                             \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(inst, ports)),                                    \
+	    (DT_CHILD(DT_INST_CHILD(inst, ports), port_##pid)), (DT_INST_CHILD(inst, port_##pid)))
+
+/**
+ * @brief Get a port node from its id
+ *
+ * Given a device instance number, return a port node specified by its ID.
+ * It handles various ways of how a port could be defined. For example:
+ *
+ * @code{.c}
+ *	DT_INST_PORT_BY_ID(inst, 0)
+ * @endcode
+ *
+ * The above snippet returns the @c port@0 or @c port node for each case listed below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		ports {
+ *			#address-cells = <1>;
+ *			#size-cells = <0>;
+ *			port@0 {
+ *				reg = <0x0>;
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		#address-cells = <1>;
+ *		#size-cells = <0>;
+ *		port@0 {
+ *			reg = <0x0>;
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @return port node associated with @p pid
+ */
+#define DT_INST_PORT_BY_ID(inst, pid)                                                              \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_PORT_BY_ID(inst, pid)),                                \
+		(_DT_INST_PORT_BY_ID(inst, pid)), (DT_INST_CHILD(inst, port)))
+
+/**
+ * @brief Helper for @ref DT_INST_ENDPOINT_BY_ID
+ *
+ * This behaves the same way as @ref DT_INST_PORT_BY_ID but does not work if there is only
+ * a single endpoint without address.
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @param eid endpoint ID
+ * @return endpoint node associated with @p eid and @p pid
+ */
+#define _DT_INST_ENDPOINT_BY_ID(inst, pid, eid)                                                    \
+	DT_CHILD(DT_INST_PORT_BY_ID(inst, pid), endpoint_##eid)
+
+/**
+ * @brief Get an endpoint node from its id and its parent port id
+ *
+ * Given a device instance number, a port ID and an endpoint ID, return the endpoint node.
+ * It handles various ways of how a port and an endpoint could be defined as described in
+ * @ref DT_INST_PORT_BY_ID and below. For example:
```

Comment:
It's recommended to replace `For example` with this
```
 * Example devicetree fragment:
```
`devicetree.h` currently almost all use this, and I don’t want to see their inconsistency, this is necessary for the neatness of the file.

Reply:
Will do, thanks !